conf.d/consul.yaml.example

```diff
@@ -23,11 +23,25 @@ instances:
       # Whether to perform checks against the Consul service Catalog
       catalog_checks: yes
 
+      # Whether to enable self leader checks. Each agent with this enabled will
+      # watch for itself to become the leader and will emit an event when that
+      # happens. It is safe/expected to enable this on all nodes in a consul
+      # cluster since only the new leader will emit the (single) event. This
+      # flag takes precedence over new_leader_checks.
+      self_leader_check: yes
```

Comment:
As mentioned earlier, I like this a lot better than what we were previously doing. However this would modify the check behavior and could throw some people off. Don't make any changes to this (or the code) just yet - but we might have to (maybe in a quick separate PR).